packages/host/app/services/network.ts

```diff
@@ -0,0 +1,84 @@
+import Service, { service } from '@ember/service';
+
+import {
+  VirtualNetwork,
+  baseRealm,
+  fetcher,
+  RunnerOpts,
+  authorizationMiddleware,
+} from '@cardstack/runtime-common';
+
+import config from '@cardstack/host/config/environment';
+
+import { shimExternals } from '../lib/externals';
+
+import type LoaderService from './loader-service';
+import type RealmService from './realm';
+
+const isFastBoot = typeof (globalThis as any).FastBoot !== 'undefined';
+
+function getNativeFetch(): typeof fetch {
+  if (isFastBoot) {
+    let optsId = (globalThis as any).runnerOptsId;
+    if (optsId == null) {
+      throw new Error(`Runner Options Identifier was not set`);
+    }
+    let getRunnerOpts = (globalThis as any).getRunnerOpts as (
+      optsId: number,
+    ) => RunnerOpts;
+    return getRunnerOpts(optsId)._fetch;
+  } else {
+    return fetch;
+  }
+}
+
+export default class NetworkService extends Service {
+  @service declare fastboot: { isFastBoot: boolean };
+  @service declare realm: RealmService;
+  @service declare loaderService: LoaderService;
+
+  virtualNetwork = this.makeVirtualNetwork();
```

Comment:
i think we want to clear this state after logout? if so this service should use the `ResetService` to register a callback that is invoked when the user logs out